megatron/rl/rl_utils.py

```diff
@@ -1708,16 +1816,31 @@ def prepare_data_for_update(
             for batch_idx, (b_trajs, b_posids) in enumerate(data_iter):
                 # Get attention mask slice
                 if compute_attention_mask is not None:
-                    start_idx = batch_idx * args.micro_batch_size
+                    start_idx = batch_idx * logprobs_batch_size
                     end_idx = min(
-                        start_idx + args.micro_batch_size, compute_attention_mask.shape[0]
+                        start_idx + logprobs_batch_size, compute_attention_mask.shape[0]
                     )
                     b_attn_mask = compute_attention_mask[start_idx:end_idx].cuda()
                 else:
                     b_attn_mask = None
 
+                b_trajs = b_trajs.cuda()
```

Comment:
Let's wrap this into an inner function to avoid copypasting the same logic twice, please?